sdk/rust/crates/dagger-sdk/src/core/cli_session.rs

```diff
@@ -50,6 +50,8 @@ impl InnerCliSession {
             args.extend(["--project".into(), abs_path.to_string_lossy().to_string()])
         }
 
+        args.extend(["--label".into(), "dagger.io/sdk.name:rust".into()]);
```

Comment:
I don't know how to get SDK version using Rust. (I never use rust also) 😢 . Do you have any advice?

Reply:
The sdk version is stored here: https://github.com/dagger/dagger/blob/1d902d3f2e3bcc20765683602dbdff214ed012ac/sdk/rust/crates/dagger-sdk/src/core/mod.rs#L1

should be as simple as: 

```suggestion
        args.extend(["--label".into(), format!("dagger.io/sdk.name:rust:version:{}", crate::DAGGER_ENGINE_VERSION).into()]);
```

Reply:
I don't know the exact format of it, but I think you get the gist of it.

Reply:
As far as I know, the sdk version should use the library version not engine version.

in the python sdk, they use importlib to get the version. https://github.com/dagger/dagger/blob/main/sdk/python/src/dagger/engine/cli.py#L24

In go sdk, https://github.com/dagger/dagger/blob/main/sdk/go/internal/engineconn/session.go#L47

Reply:
> I don't know the exact format of it, but I think you get the gist of it.

There is a new label with annotate the sdk version.

Ref1 https://github.com/dagger/dagger/blob/main/sdk/go/internal/engineconn/session.go#L82
Ref2 https://github.com/dagger/dagger/blob/main/sdk/CONTRIBUTING.md#automatic-provisioning. The label info is in TODO in this section

Reply:
Ah right. There is this thing here 

`let version = env!("CARGO_PKG_VERSION");`

https://doc.rust-lang.org/cargo/reference/environment-variables.html

Not quite sure what that would do when consumed. But it could be useful on the publishing step.

Reply:
Thank you! Let me try tonight. :)

Reply:
@kjuulh I tested and it looks works. You can see it at https://github.com/dagger/dagger/pull/5182/files#diff-085b907847a5ba3189f3ba233672dbab52b85b57c140146d62c74c8eda75e75aR54-R57. 🙇

Reply:
Great let's get it merged I will test it in one of my downstream apps to make sure it is set properly. I will probably first get to it tomorrow so let's get it merged now